pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -383,20 +382,33 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env model.En
 				}
 
 				listenerMapKey = fmt.Sprintf("%s:%d", listenAddress, servicePort.Port)
-				if n, exists := listenerMap[listenerMapKey]; exists {
-					log.Warnf("Multiple TCP listener definitions for %s %v %s", listenerMapKey, n, service.Hostname)
-					continue
+				var exists bool
+				if currentListener, exists = listenerMap[listenerMapKey]; exists {
+					// Check if this is HTTPS port collision. If so, we can use SNI to differentiate
+					if servicePort.Protocol != model.ProtocolHTTPS {
+						log.Warnf("Multiple TCP listener definitions for %s %v %s", listenerMapKey, currentListener, service.Hostname)
+						continue
+					}
 				}
-				listenerOpts.filterChainOpts = []*filterChainOpts{{
+				filterChainOption := &filterChainOpts{
 					networkFilters: buildOutboundNetworkFilters(clusterName, addresses, servicePort),
-				}}
+				}
+
+				// TODO (@rshriram): This is not sufficient. There are other TCP protocols that use SNI, that need to be tackled.
```

Comment:
We need to separate SNI/TLS from Protocol, this is only going to bite us more later. We really should be using Gateway, DestinationRule, and mesh default config to decide when to wire up SNI matching, right?

Reply:
No actually. None of these apply to the outbound listener on a sidecar. Outbound listeners are either tcp (https inclusive) or http. I am merely separating the tcp out on the outbound path so that we can avoid the issue around exclusive use of a port 443, by a single external service (ir one without a service ip). Iow with this we can support https://google and https://yahoo